presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -212,6 +212,12 @@ public static boolean canCoerce(List<? extends Type> actualTypes, List<Type> exp
         return true;
     }
 
+    private static boolean isDateTimeType(String type)
```

Comment:
Maybe name it like `getVarcharSubtypes`, and make it return the list of types including `JsonPath` and etc. Than just use that list in both places `canCastTypeBase` and `getCommonSuperTypeForTypesWithLongParameters`